src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java

```diff
@@ -1,24 +1,38 @@
 package com.bootcamp.demo.dao;
 
+import com.bootcamp.demo.mapper.DocumentToCardMapper;
+import com.bootcamp.demo.model.Card;
+import com.google.api.core.ApiFuture;
 import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.FirebaseOptions;
 import com.google.firebase.cloud.FirestoreClient;
-import org.springframework.context.annotation.Bean;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
 import javax.annotation.PostConstruct;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
 
 import static java.lang.System.getProperty;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 @Repository
 public class FirestoreDaoImpl implements FirestoreDao{
     private Firestore firestoreDB;
+    DocumentToCardMapper mapper;
```

Comment:
Please make this field `private final` so it respects the encapsulation principle.

Reply:
Fixed